runtime/java/src/common/Debug.java

```diff
@@ -0,0 +1,1198 @@
+
+                //used to access parents that forwarded to this node
+                case "backtrack": case "backwards": case "b":
+                    //backtrack should not have any arguments
+                    if (userInputList.length != 1) {
+                        System.out.println("invalid, correct usage: backtrack<>");
+                        break;
+                    }
+
+                    //try to move backwards
+                    if(backtrack() == -1)
+                        break;
+
+                    //Update the json and html and terminal to the new node
+                    updateDisplay();
+                    break;
+
+                //Used to specify what displays the user want to see
+                case "toggle":
+                    String toggleInput = "";
+
+                    //If the user does not provide a child we should list them out
+                    if (userInputList.length == 1) {
+                        System.out.println("Which toggle?");
+                        displayArray(toggleChoices);
+                        System.out.print(">DEBUGGER-PROMPT$");
+                        toggleInput = inp.nextLine();
+                    }
+                    
+                    //Otherwise they should have typed one in
+                    else if (userInputList.length == 2){
+                        toggleInput = userInputList[1];
+                    } 
+
+                    //We do not expect more than 2 arguments for toggle 
+                    else{
+                        System.out.println("invalid, correct usage: toggle <toggle name>");
+                        break;
+                    }
+
+                    //Activate the toggle
+                    toggle(toggleInput);
+                    break;
+
+                //used to display the production name on the terminal
+                case "name": 
+                    
+                    //name should not have any arguments
+                    if (userInputList.length != 1) {
+                        System.out.println("invalid, correct usage: name");
+                        break;
+                    }
+
+                    //Print the name
+                    printName(currentNode);
+                    break;
+
+                //used to jump to a specific equation
+                case "equation": case "eq": 
+                    String attributeNameInput = ""; 
+
+                    //If the user does not provide a attribute we should list them out
+                    if (userInputList.length == 1) {
+                        List<String> attributeListView = allAttributesList(currentNode);
+                        String[] attributeArrayView = attributeListView.toArray(new String[attributeListView.size()]);
+                        System.out.println("Which attribute?");
+                        displayArray(attributeArrayView);
+                        System.out.print(">DEBUGGER-PROMPT$");
```

Comment:
It's a bit confusing that the prompt is the same for choosing the attribute.  Could just have `"Which attribute? "` as the prompt.